dbms/src/Functions/FunctionsRegexp.h

```diff
@@ -973,15 +1114,373 @@ class FunctionStringRegexp : public FunctionStringRegexpBase
+            Int64 ret_op = RetOpT::isConst() ? ret_op_const_val : get_ret_op_func(ret_op_container, 0);
+            String match_type = match_type_param.getString(0);
+
+            Regexps::Regexp regexp(addMatchTypeForPattern(pat, match_type, collator), flags);
+            ResultType res = regexp.instr(expr.c_str(), expr.size(), pos, occur, ret_op);
+            res_arg.column = res_arg.type->createColumnConst(col_size, toField(res));
+            return;
+        }
+
+        // Initialize result column
+        auto col_res = ColumnVector<ResultType>::create();
+        typename ColumnVector<ResultType>::Container & vec_res = col_res->getData();
+        vec_res.resize(col_size, 0);
+
+        constexpr bool has_nullable_col = ExprT::isNullableCol() || PatT::isNullableCol() || PosT::isNullableCol() || OccurT::isNullableCol() || RetOpT::isNullableCol() || MatchTypeT::isNullableCol();
+
+#define GET_POS_VALUE(idx) \
+    do \
+    { \
+        if constexpr (PosT::isConst()) \
+            pos = pos_const_val; \
+        else \
+            pos = get_pos_func(pos_container, idx); \
+    } while (0);
+
+#define GET_OCCUR_VALUE(idx) \
+    do \
+    { \
+        if constexpr (OccurT::isConst()) \
+            occur = occur_const_val; \
+        else \
+            occur = get_occur_func(occur_container, idx); \
+    } while (0);
+
+#define GET_RET_OP_VALUE(idx) \
+    do \
+    { \
+        if constexpr (RetOpT::isConst()) \
+            ret_op = ret_op_const_val; \
+        else \
+            ret_op = get_ret_op_func(ret_op_container, idx); \
+    } while (0);
+
+        StringRef expr_ref;
+        String pat;
+        Int64 pos;
+        Int64 occur;
+        Int64 ret_op;
+        String match_type;
+
+        // Start to execute instr
+        if (canMemorize<PatT, MatchTypeT>())
+        {
+            // Codes in this if branch execute instr with memorized regexp
+
+            const auto & regexp = memorize(pat_param, match_type_param, collator);
+            if constexpr (has_nullable_col)
+            {
+                // Process nullable columns with memorized regexp
+                auto nullmap_col = ColumnUInt8::create();
+                typename ColumnUInt8::Container & null_map = nullmap_col->getData();
+                null_map.resize(col_size);
+
+                for (size_t i = 0; i < col_size; ++i)
+                {
+                    if (expr_param.isNullAt(i) || pos_param.isNullAt(i) || occur_param.isNullAt(i) || ret_op_param.isNullAt(i))
+                    {
+                        null_map[i] = 1;
+                        continue;
+                    }
+                    null_map[i] = 0;
+                    expr_param.getStringRef(i, expr_ref);
+                    GET_POS_VALUE(i)
+                    GET_OCCUR_VALUE(i)
+                    GET_RET_OP_VALUE(i)
+                    vec_res[i] = regexp->instr(expr_ref.data, expr_ref.size, pos, occur, ret_op);
```

Comment:
Why don't check if `regexp` is nullptr?

Reply:
> Why don't check if `regexp` is nullptr?

we can ensure regexp is not nullptr here